### PR TITLE
Replace Search in Header

### LINKED ISSUE #25: Replace search in header with new search component
In our header component, we need to replace the input with our new search component

### PR DESCRIPTION
**Pull request by jonathanguven**
closes #25

### CODE REVIEW

File: src/routes/+layout.server.ts [hunk 1]
```diff
@@ -1,10 +1,23 @@
-import type { RequestEvent } from '@sveltejs/kit';
+import { type RequestEvent } from '@sveltejs/kit';
 import type { PageServerLoad } from './$types';
+import { zod } from 'sveltekit-superforms/adapters';
+import { z } from 'zod';
+import { superValidate } from 'sveltekit-superforms';
+
+const schema = z.object({
```
Comment (gursheyss) [inline src/routes/+layout.server.ts:7]: We need to move this to another file and export the schema so we can use it as a type in props

File: src/lib/components/Header.svelte [hunk 2]
```diff
@@ -1,41 +1,16 @@
...
-			class="input input-bordered input-accent mx-4 {!showSearch
-				? 'hidden'
-				: ''} h-10 grow justify-between gap-x-2 md:max-w-[35%] lg:flex"
-		>
-			<input class="max-w-full grow" type="text" placeholder="Search" bind:this={searchBox} />
-			<Search class="hidden lg:flex" />
-		</label>
-	{/if}
+	<a href="/" class="btn btn-ghost text-2xl md:flex">Spartan Compass</a>
+	<Search {formData} />
```
Comment (gursheyss) [inline src/lib/components/Header.svelte:13]: We don't want to show this on the home page

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -1,15 +1,10 @@
 <script lang="ts">
-	import Search from '$lib/components/Search.svelte';
 	let { data } = $props();
 	let user = $derived(data.user);
-	let formData = $derived(data.form);
 </script>
 
-<div class="flex h-screen items-center justify-center">
-	<Search {formData} />
```
Comment (gursheyss) [inline src/routes/+page.svelte:9]: This should still show on the home page

File: src/lib/components/Header.svelte [hunk 4]
```diff
@@ -1,41 +1,16 @@
...
-	let searchBox: HTMLInputElement | undefined = $state();
-
-	async function toggleSearch() {
-		showSearch = !showSearch;
-		if (showSearch && searchBox) {
-			await tick();
-			searchBox.focus();
-		}
-	}
+	let { user, formData } = $props();
```
Comment (gursheyss) [inline src/lib/components/Header.svelte:8]: This needs to be typed

**Review by gursheyss — CHANGES_REQUESTED** [review]

File: src/routes/+layout.server.ts (hunk 1 above)
Comment (jonathanguven) [reply src/routes/+layout.server.ts]: idk if i did this right

**Review by gursheyss — APPROVED** [review]

**Pull request merged**